graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/servlet/auth/GraphQLServletSecurityValidator.java

```diff
@@ -0,0 +1,144 @@
+import graphql.parser.Parser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.security.auth.Subject;
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.NameCallback;
+import javax.security.auth.callback.PasswordCallback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.List;
+
+import static graphql.language.OperationDefinition.Operation.MUTATION;
+import static graphql.language.OperationDefinition.Operation.QUERY;
+import static graphql.language.OperationDefinition.Operation.SUBSCRIPTION;
+import static org.osgi.service.http.HttpContext.AUTHENTICATION_TYPE;
+import static org.osgi.service.http.HttpContext.REMOTE_USER;
+
+public class GraphQLServletSecurityValidator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GraphQLServletSecurityValidator.class);
+
+    private final Parser parser;
+
+    public GraphQLServletSecurityValidator() {
+        parser = new Parser();
+    }
+
+    public boolean validate(String query, String operationName, HttpServletRequest req, HttpServletResponse res) throws IOException {
+        if (isPublicOperation(query)) {
+            return true;
+        } else if (req.getHeader("Authorization") == null) {
+            res.addHeader("WWW-Authenticate", "Basic realm=\"karaf\"");
+            res.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+            return false;
+        }
+
+        if (isAuthenticatedUser(req)) {
+            return true;
+        } else {
+            res.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+            return false;
+        }
+    }
+
+    private boolean isPublicOperation(String query) {
+        final Document queryDoc = parser.parseDocument(query);
+        final Definition<?> def = queryDoc.getDefinitions().get(0);
+        if (def instanceof OperationDefinition) {
+            OperationDefinition opDef = (OperationDefinition) def;
+            if ("IntrospectionQuery".equals(opDef.getName())
+                    || SUBSCRIPTION.equals(opDef.getOperation())) {
+                // allow subscriptions and introspection query
+                return true;
+            }
+
+            List<Node> children = opDef.getSelectionSet().getChildren();
+            final Field cdp = (Field) children.stream().filter((node) -> {
+                return (node instanceof Field) && "cdp".equals(((Field) node).getName());
+            }).findFirst().orElse(null);
+            if (cdp == null) {
+                // allow not a cdp namespace
+                return true;
+            }
+
+            String allowedNodeName = "";
+            if (QUERY.equals(opDef.getOperation())) {
+                allowedNodeName = "getProfile";
```

Comment:
This look dangerous, because the Unomi ticket description mentioned:
**Retrieve current profile**
But looking at the current method, anybody could read the profile of the others. since the getProfile query can take a profileId as parameter.